src/clients/api/queries/getMarketHistory/index.ts

```diff
@@ -0,0 +1,78 @@
+import { VError } from 'errors';
+import { MarketSnapshot, VToken } from 'types';
+import { restService } from 'utilities';
+
+const ENTRIES_HOUR = 2;
+export const ENTRIES_DAY = ENTRIES_HOUR * 24;
+const ENTRIES_WEEK = ENTRIES_DAY * 7;
+const ENTRIES_30_DAYS = ENTRIES_DAY * 30;
+const ENTRIES_YEAR = ENTRIES_DAY * 365;
+
+export type MarketHistoryTypes = '1 day' | '1 week' | '1 month' | '1 year';
+
+export interface GetMarketHistoryResponse {
+  asset: string;
+  data: MarketSnapshot[];
+  updatedAt: string;
+}
+
+export interface GetMarketHistoryInput {
+  vToken: VToken;
+  limit?: number;
+  type?: MarketHistoryTypes;
+}
+
+export type GetMarketHistoryOutput = {
+  marketSnapshots: MarketSnapshot[];
+};
+
+const getMarketHistory = async ({
+  vToken,
+  limit = 30,
+  type = '1 month',
+}: GetMarketHistoryInput): Promise<GetMarketHistoryOutput> => {
+  let endpoint = `/market_history/graph?asset=${vToken.address}&version=v2`;
+
+  if (limit) {
+    endpoint += `&limit=${limit}`;
+  }
+
+  const response = await restService<GetMarketHistoryResponse>({
+    endpoint,
+    method: 'GET',
+  });
+
+  // @todo Add specific api error handling
+  if ('result' in response && response.result === 'error') {
+    throw new VError({
+      type: 'unexpected',
+      code: 'somethingWentWrong',
+      data: { message: response.message },
+    });
+  }
+
+  let sampleSize = response.data?.data.data.length || 0;
```

Comment:
yeah we can definitely improve this just not in the scope of the initial isolated lending release.

We first need to implement a sustainable versioning pattern and then refactor all our success and error responses to be cleaner and more useful. All of these items are on the roadmap

Reply:
Well, that certainly looks redundant, it goes like this:

1st data is a frontend wrapper.
2nd data is how the backend returns all payloads.
3rd data is an actual field for this model in the database.

Number 1 is fine. Number 3 can be renamed to something like `result` instead, but that wouldn't really remove the nesting, it would just look different. Number 2 can be removed, yes, but removing it for this specific endpoint alone wouldn't be that nice. It would be a better idea to refactor how the API as a whole returns data, but that is a larger effort.

Reply:
I see, thanks for the extra info. Let's keep things as they are for now